components/server/src/auth/jwt.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import * as jsonwebtoken from "jsonwebtoken";
+import { Config } from "../config";
+import { inject, injectable } from "inversify";
+import { log } from "@gitpod/gitpod-protocol/lib/util/logging";
+
+const algorithm: jsonwebtoken.Algorithm = "RS512";
+
+@injectable()
+export class AuthJWT {
+    @inject(Config) protected config: Config;
+
+    async sign(subject: string, payload: object | Buffer, expiresIn: string = `${24 * 7}h`): Promise<string> {
+        const opts: jsonwebtoken.SignOptions = {
```

Comment:
Q: would it be of any value to add the "iat" (Issued At) claim? That basically translates to "Logged in at". 
Can be added later, just thinking out loud.

Reply:
The token already contains IAT (auto-added by the library) (if you decode the issued token)

Reply:
Indeed 🤦🏻 
Plain overlook.